### PR TITLE
Add error handling to OTPilot

### PR DESCRIPTION
**Pull request by osenft**
This PR builds on top of https://github.com/google/tock-on-titan/pull/183.

```
----------------------
`make prtest` summary:
----------------------
git rev-parse HEAD
5199c0083d9eb1df1820a40a308279dee415f74d
git status
On branch otpilot-error-handling
Your branch is up to date with 'origin/otpilot-error-handling'.

nothing to commit, working tree clean
```

### CODE REVIEW

**Comment by osenft** [comment]
I rebased this PR on top of #183 now. PTAL.

File: userspace/otpilot/src/main.rs [hunk 1]
```diff
@@ -236,51 +278,40 @@ impl<'a> SpiProcessor<'a> {
...
-            return Err(FromWireError::OutOfRange);
-        }
+        header.to_wire(&mut tx_cursor)?;
         if header.get_opcode().has_dummy_byte() {
             // Skip dummy byte
-            if let Err(why) = tx_cursor.write_bytes(&[1; 0]) {
-                writeln!(console, "Host: Could not store dummy byte: {:?}", why);
-                return Err(FromWireError::OutOfRange);
-            }
+            tx_cursor.write_bytes(&[1; 0])
```
Comment (jrvanwhy) [inline userspace/otpilot/src/main.rs]: Was this supposed to be a length-1 array containing a zero byte rather than an empty slice?

Comment (osenft) [reply userspace/otpilot/src/main.rs]: No, we're writing a 1 byte array with content "0x0". That's the "dummy" byte that needs to be sent. Did I get the syntax wrong?

Comment (jrvanwhy) [reply userspace/otpilot/src/main.rs]: Yes, a 1-byte array with content "0x0" is `&[0; 1]`.

Comment (osenft) [reply userspace/otpilot/src/main.rs]: Oops ... jeez. What a bug. Thanks for noticing that. That would have caused some head-scratching at some point! Fixed in separate commit in same PR.

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**